### PR TITLE
Merging development into Main

### PR DESCRIPTION
**Pull request by RidwanHamid501**
Merging development into Main

### CODE REVIEW

File: dashboard/pages/Subscribe.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+    """Execute a query and return results if applicable"""
+    try:
+        conn, cursor = get_connection()
+        cursor.execute(query, params)
+        if fetch_one:
+            return cursor.fetchone()
+        conn.commit()
+        conn.close()
+        return None
+    except psycopg2.DatabaseError as e:
```
Comment (keoghrmj) [inline dashboard/pages/Subscribe.py:62]: nice use of an execute function that can be reused a lot.

File: dashboard/pages/Trend_Related_Terms.py [hunk 2]
```diff
@@ -0,0 +1,221 @@
+""""dfsdf"""
```
[the remark this answers is not in the document]
Comment (keoghrmj) [reply dashboard/pages/Trend_Related_Terms.py:1]:
```suggestion
""""Related terms page on the dashboard"""
```

File: dashboard/pages/Trend_Related_Terms.py [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+        st.write("Explore the related terms:")
+        graph1, _, graph2 = st.columns([7, 1, 7])
+        with graph1:
+            clicked_node1 = network_graph(selected_keywords[0], cursor)
+            # if clicked_node1:
+            # st.session_state.clicked_nodes.append(clicked_node1)
+            # st.rerun()
+        with graph2:
+            clicked_node2 = network_graph(selected_keywords[1], cursor)
+            # if clicked_node2:
```
Comment (Kurt812) [inline dashboard/pages/Trend_Related_Terms.py:206]: Some comments here

File: dashboard/pages/Trend_Related_Terms.py [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+        </div>
+        """,
+        unsafe_allow_html=True
+    )
+
+
+def initialize_trend_request() -> TrendReq:
+    """Initialize and return a TrendReq object."""
+    return TrendReq()
+
```
Comment (keoghrmj) [inline dashboard/pages/Trend_Related_Terms.py:43]: is there much point of this function?

File: dashboard/pages/Trend_Related_Terms.py [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+                st.rerun()
+
+    elif len(selected_keywords) == 2:
+        st.write("Explore the related terms:")
+        graph1, _, graph2 = st.columns([7, 1, 7])
+        with graph1:
+            clicked_node1 = network_graph(selected_keywords[0], cursor)
+            # if clicked_node1:
+            # st.session_state.clicked_nodes.append(clicked_node1)
+            # st.rerun()
```
Comment (keoghrmj) [inline dashboard/pages/Trend_Related_Terms.py:203]: why is there commented code?

File: dashboard/pages/Trend_Related_Terms.py [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+        with graph2:
+            clicked_node2 = network_graph(selected_keywords[1], cursor)
+            # if clicked_node2:
+            # st.session_state.clicked_nodes.append(clicked_node2)
+            # st.rerun()
```
Comment (keoghrmj) [inline dashboard/pages/Trend_Related_Terms.py:208]:
```suggestion
```

**Pull request merged**